HPCC-25006 Add RoxieQueueFilter to TpLogicalClusterQuery

wangkx opened the pull request:
This filter is used to filter targets in a cloud environment.
The filter has three options: All, QueriesOnly, and WorkunitsOnly
(default to All). 

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

wangkx committed:
HPCC-25006 Add DeployOnlyFilter to TpLogicalClusterQuery

This filter is used to filter targets in a cloud environment.
The filter has three options: All, DeployOnly, and NotDeployOnly.
To filter out the DeployOnly targets, the filter should be set as
NotDeployOnly.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

HPCCSmoketest commented:
Process of PR-14334, label: h24945 starts now.
The reason of this test is: Forced to re-test.
Commit ID: 14bf2efa8e18549a4336327051724ce9af0eff65
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-196.ca-central-1.compute.internal

wangkx commented:
@afishbeck please review

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-196.ca-central-1.compute.internal
Sha: 14bf2efa8e18549a4336327051724ce9af0eff65
Build: success
Milestone:Install hpccsystems-platform-community_7.12.11-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |146 |146 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |23 sec  (00:00:23) |
|setup (thor) |11 |11 |0 |40 sec  (00:00:40) |
|setup (roxie) |11 |11 |0 |17 sec  (00:00:17) |
|test (hthor) |936 |936 |0 |305 sec  (00:05:05) |
|test (thor) |846 |846 |0 |655 sec  (00:10:55) |
|test (roxie) |1015 |1015 |0 |352 sec  (00:05:52) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  600 sec (00:10:00) |  104 sec (00:01:44) |   21 sec (00:00:21) |   19 sec (00:00:19) | 1628 sec (00:27:08) |   18 sec (00:00:18) | 2398 sec (00:39:58) |

afishbeck commented on esp/scm/ws_topology.ecm:
```diff
@@ -17,6 +17,13 @@
+    All("All"),
+    DeployOnly("DeployOnly"),
+    NotDeployOnly("NotDeployOnly")
+};
```
@richardkchapman Just want to make sure we have the intention and terminology right here.
In many places deploy some ecl is to compile the workunit.. that's not the same thing as publishing a query.  So there are 3 actions.
1. deploy / compile a workunit.
2. run a workunit
3. publish a query.
ECL could be compiled for the roxie (deployed as workunits) but wouldn't end up callable as queries yet.

If the focus is on the workunit it could be:
All, Runnable, CompileOnly.

If the focus is on how things are run it could be:
All, WorkunitsOnly, QueriesOnly

In both cases we really allow publishing/compiling to other cluster types so in practice it would just be All and either CompileOnly or QueriesOnly.

wangkx replied:
I guess we are trying to separate 2 groups of roxies here: the roxies for running (compile and run) workunits and the roxies for publishing queries. If so, we should change to 'All, WorkunitsOnly, QueriesOnly'. But, I could be wrong.

afishbeck replied:
Makes sense to me.

richardkchapman commented:
I think this is back with @wangkx

wangkx committed:
HPCC-25025 Rename the DeployOnly filter

Rename the DeployOnlyFilter to RoxieQueueFilter.
Rename the DeployOnly to QueriesOnly..
Rename the NonDeployOnly to WorkunitsOnly.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

HPCCSmoketest commented:
Process of PR-14334, label: h24945 starts now.
The reason of this test is: Forced to re-test.
Commit ID: a1230ac8ba21e7026d3e6913b1d41955080288c1
Estimated completion time is ~0.75 hour(s)
centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-46.ca-central-1.compute.internal

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-46.ca-central-1.compute.internal
Sha: a1230ac8ba21e7026d3e6913b1d41955080288c1
Build: success
Milestone:Install hpccsystems-platform-community_7.12.13-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |145 |145 |0 |0 |0 |52 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |1 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |5 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |8 |8 |0 |21 sec  (00:00:21) |
|setup (thor) |8 |8 |0 |41 sec  (00:00:41) |
|setup (roxie) |11 |11 |0 |22 sec  (00:00:22) |
|test (hthor) |941 |941 |0 |304 sec  (00:05:04) |
|test (thor) |852 |852 |0 |775 sec  (00:12:55) |
|test (roxie) |1020 |1020 |0 |352 sec  (00:05:52) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    9 sec (00:00:09) |  600 sec (00:10:00) |  104 sec (00:01:44) |   21 sec (00:00:21) |   16 sec (00:00:16) | 1751 sec (00:29:11) |   16 sec (00:00:16) | 2517 sec (00:41:57) |

wangkx commented:
@afishbeck please review

afishbeck reviewed (approved):
@wangkx looks good.
